src/test/java/org/prebid/server/bidder/coinzilla/CoinzillaBidderTest.java

```diff
@@ -0,0 +1,199 @@
+package org.prebid.server.bidder.coinzilla;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.iab.openrtb.request.Banner;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.response.Bid;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.netty.handler.codec.http.HttpHeaderValues;
+import io.vertx.core.MultiMap;
+import org.junit.Before;
+import org.junit.Test;
+import org.prebid.server.VertxTest;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.HttpResponse;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.UnaryOperator;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static java.util.function.UnaryOperator.identity;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.assertj.core.api.Assertions.tuple;
+
+public class CoinzillaBidderTest extends VertxTest {
+
+    private static final String ENDPOINT_URL = "https://coinzilla.com";
+
+    private CoinzillaBidder coinzillaBidder;
+
+    @Before
+    public void setUp() {
+        coinzillaBidder = new CoinzillaBidder(ENDPOINT_URL, jacksonMapper);
+    }
+
+    @Test
+    public void creationShouldFailOnInvalidEndpointUrl() {
+        assertThatIllegalArgumentException().isThrownBy(() ->
+                new CoinzillaBidder("invalid_url", jacksonMapper));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorIfNoImpsPresent() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(bidRequestBuilder ->
+                bidRequestBuilder.imp(emptyList()), identity());
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = coinzillaBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1)
```

Comment:
`hasSize` redundant

Reply:
To my mind `.hasSize` with `.containsExactly` is fine. While if we have `.containsOnly` it is 100% redundant and we are having this all over the codebase.

Reply:
You decide what to do, but anyway in `.containsExactly ` body we can see what exactly we expect.